### PR TITLE
Add test logic to account for multiple ports per PortChannel

### PR DESCRIPTION
**Pull request by justin-wong-ce**
### Description of PR

Summary: Add test logic to account for port channel setups where there are multiple ports associated to the port channel.
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
    - [ ] Skipped for non-supported platforms
- [x] Test case improvement

### Back port request
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [ ] 202411
- [ ] 202505
- [x] 202511

#### What is the motivation for this PR?
The test is failing on HwSKUs where there are multiple ports associated with a port channel.

#### How did you do it?
Add test logic to account for the possibility that there can be multiple ports associated with a port channel.

#### How did you verify/test it?
Test no longer fails on devices where there are multiple ports associated to a single port channel.

### CODE REVIEW

File: tests/snmp/test_snmp_interfaces.py [hunk 1]
```diff
@@ -49,10 +49,10 @@ def get_interfaces(duthost, tbinfo):
     rif_counters = parse_rif_counters(duthost.command("show interfaces counters rif")["stdout_lines"])
     for interface in rif_counters:
         if 'Eth' in interface:
-            return interface, interface
+            return [interface], interface
         else:
             mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
-            return mg_facts["minigraph_portchannels"][interface]['members'][0], interface
+            return mg_facts["minigraph_portchannels"][interface]['members'], interface
```
Comment (qiluo-msft) [inline tests/snmp/test_snmp_interfaces.py:58]: could you handle corner case that zero member port channel? Do you want to test on it, or skip and select another one?

File: tests/snmp/test_snmp_interfaces.py [hunk 2]
```diff
@@ -264,10 +264,10 @@ def verify_snmp_counter(duthost, localhost, creds_all_duts, hostip, mg_facts, ri
         logger.info(f"ifOutDiscards value is {rif_snmp_facts['ifOutDiscards']} but must be "
                     f"{int(rif_counters[rif_interface]['tx_err']) + int(port_counters['tx_drp'])}")
         return False
-    if int(rif_snmp_facts['ifInErrors']) != COUNTER_VALUE:
+    if int(rif_snmp_facts['ifInErrors']) != COUNTER_VALUE * num_port_intfs:
         logger.info(f"ifInErrors value is {rif_snmp_facts['ifInErrors']} but must be {COUNTER_VALUE}")
```
Comment (qiluo-msft) [inline tests/snmp/test_snmp_interfaces.py]: Could you update logging message to reflect code change?

File: tests/snmp/test_snmp_interfaces.py [hunk 3]
```diff
@@ -264,10 +264,10 @@ def verify_snmp_counter(duthost, localhost, creds_all_duts, hostip, mg_facts, ri
...
         return False
-    if int(rif_snmp_facts['ifOutErrors']) != COUNTER_VALUE:
+    if int(rif_snmp_facts['ifOutErrors']) != COUNTER_VALUE * num_port_intfs:
         logger.info(f"ifOutErrors value is {rif_snmp_facts['ifOutErrors']} but must be {COUNTER_VALUE}")
```
Comment (qiluo-msft) [inline tests/snmp/test_snmp_interfaces.py]: Could you update logging message to reflect code change?

File: tests/snmp/test_snmp_interfaces.py (hunk 1 above)
Comment (justin-wong-ce) [reply tests/snmp/test_snmp_interfaces.py:58]: If there is no RIF interface nor port channel interfaces, then I think the test will have to be skipped

File: tests/snmp/test_snmp_interfaces.py [hunk 4]
```diff
@@ -49,10 +49,13 @@ def get_interfaces(duthost, tbinfo):
...
         if 'Eth' in interface:
-            return interface, interface
+            return [interface], interface
         else:
             mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
-            return mg_facts["minigraph_portchannels"][interface]['members'][0], interface
+            if interface not in mg_facts["minigraph_portchannels"].keys():
+                continue
+            return mg_facts["minigraph_portchannels"][interface]['members'], interface
+    pytest.skip("No RIF interfaces nor Portchannels, skiping the test")
```
Comment (StormLiangMS) [inline tests/snmp/test_snmp_interfaces.py]: **Zero-member PortChannel not guarded**: If a PortChannel exists in `mg_facts` but has empty members (`[]`), this returns an empty list making `num_port_intfs = 0`. All counter assertions then check against `0`, silently masking failures. Please add a guard:
```python
members = mg_facts["minigraph_portchannels"][interface]['members']
if not members:
    continue
return members, interface
```

File: tests/snmp/test_snmp_interfaces.py [hunk 5]
```diff
@@ -49,10 +49,13 @@ def get_interfaces(duthost, tbinfo):
...
```
Comment (StormLiangMS) [inline tests/snmp/test_snmp_interfaces.py:60]: Typo: `skiping` → `skipping`, `Portchannels` → `PortChannels`

**Review by StormLiangMS — COMMENTED** [review]
@justin-wong-ce 
## Code Review

Thanks for the fix! The overall approach of returning a list of members and aggregating counters across all PortChannel member ports is correct. A few issues to address:

---

### 🐛 Issue 1: Zero-member PortChannel not guarded

In `get_interfaces()`, if a PortChannel exists in `mg_facts` but has no members (`'members': []`), the function returns an empty list. This makes `num_port_intfs = 0` and all counter assertions become `COUNTER_VALUE * 0 == 0`, silently masking failures.

**Suggestion:**
```python
members = mg_facts["minigraph_portchannels"][interface]['members']
if not members:
    continue
return members, interface
```

---

### 🐛 Issue 2: `ifInDiscards` / `ifOutDiscards` in `verify_snmp_counter` may also need `* num_port_intfs`

The PR updates `ifInErrors` and `ifOutErrors` to use `* num_port_intfs`, but the `ifInDiscards` / `ifOutDiscards` checks earlier in `verify_snmp_counter` are unchanged. Since `SAI_PORT_STAT_IF_IN_DISCARDS` and `SAI_PORT_STAT_IF_OUT_DISCARDS` are now set on **all** member ports in the loop, the SNMP aggregated value would also be `COUNTER_VALUE * num_port_intfs` for discards. Please verify and update those checks if needed.

---

### 🔤 Nit 1: Typo in skip message

```python
# Current:
pytest.skip("No RIF interfaces nor Portchannels, skiping the test")
# Fix:
pytest.skip("No RIF interfaces nor PortChannels, skipping the test")
```

---

### 🔤 Nit 2: Use `zip()` instead of `enumerate()` for clarity

```python
# Current:
for i, port_oid in enumerate(port_oids):
    logger.info(f'Set port {port_interfaces[i]} ...')

# Cleaner:
for port_interface, port_oid in zip(port_interfaces, port_oids):
    logger.info(f'Set port {port_interface} ...')
```

---

The main concern is **Issue 2** — please verify whether `ifInDiscards`/`ifOutDiscards` in `verify_snmp_counter` also need the `num_port_intfs` multiplier before merging.

**Review by StormLiangMS — APPROVED** [review]
LGTM — my previous feedback has been addressed:

- Zero-member PortChannel guard added (skips and tries next interface)
- Typo fixed (\skipping\, \PortChannels\)
- \pytest.skip()\ at end handles no-valid-interface case

The logic is sound:
1. \get_interfaces()\ now returns a **list** of port members (consistent for both Eth and PortChannel cases)
2. Counters are set on **each** member port individually
3. Aggregated counters are compared against \COUNTER_VALUE * num_port_intfs\
4. SNMP verification also uses \
um_port_intfs\ multiplier for \ifInErrors\/\ifOutErrors\

Only the VPP optional test failed — not related to this change.

**Comment by StormLiangMS** [comment]
Friendly ping — this PR has been open since Jan 23 and needs review/merge for 202511 qualification. cc @Junchao-Mellanox @xwjiang-ms @bachalla
